src/Land/Model/SoilHeatParameterizations.jl

```diff
@@ -0,0 +1,198 @@
+"""
+    SoilHeatParameterizations
+
+Functions for volumetric heat capacity, internal energy as function of temperature,
+saturated thermal conductivity, thermal conductivty, the Kersten number, and relative
+saturation are included.
+"""
+module SoilHeatParameterizations
+
+using DocStringExtensions
+
+export volumetric_heat_capacity,
+    internal_energy,
+    saturated_thermal_conductivity,
+    thermal_conductivity,
+    relative_saturation,
+    kersten_number,
+    internal_energy_liquid_water,
+    temperature_from_I
+
+
+"""
+    function temperature_from_I(
+        T_ref::FT,
+        I::FT,
+        θ_ice::FT,
+        ρ_ice::FT,
+        LH_f_0::FT,
+        cs::FT
+    ) where {FT}
+
+Computes the temperature given I and `θ_ice`.
+"""
+function temperature_from_I(
+    T_ref::FT,
+    I::FT,
+    θ_ice::FT,
+    ρ_ice::FT,
+    LH_f0::FT,
+    cs::FT,
+) where {FT}
+    T = T_ref + (I + θ_ice * ρ_ice * LH_f0) / cs
+    return T
+end
+
+
+"""
+    volumetric_heat_capacity(
+        ϴ_l::FT,
+        ϴ_i::FT,
+        c_ds::FT,
+        cp_l::FT,
+        cp_i::FT
+    ) where {FT}
+
+Compute the expression for volumetric heat capacity.
+"""
+function volumetric_heat_capacity(
+    ϴ_l::FT,
+    ϴ_i::FT,
+    c_ds::FT,
+    cp_l::FT,
+    cp_i::FT,
+) where {FT}
+    c_s = c_ds + ϴ_l * cp_l + ϴ_i * cp_i
+    return c_s
+end
+
+"""
+    internal_energy(
+        ϴ_i::FT,
+        c_s::FT,
+        T::FT,
+        T_ref::FT,
+        ρ_i::FT,
+        LH_f_0::FT
+    ) where {FT}
+Compute the expression for internal energy.
+"""
+function internal_energy(
+    ϴ_i::FT,
+    c_s::FT,
+    T::FT,
+    T_ref::FT,
+    ρ_i::FT,
+    LH_f_0::FT,
+) where {FT}
+    I = c_s * (T - T_ref) - ϴ_i * ρ_i * LH_f_0
+    return I
+end
+
+"""
+    saturated_thermal_conductivity(
+        ϴ_l::FT,
+        ϴ_i::FT,
+        κ_sat_unfrozen::FT,
+        κ_sat_frozen::FT
+    ) where {FT}
+Compute the expression for saturated thermal conductivity of soil matrix.
+"""
+function saturated_thermal_conductivity(
+    ϴ_l::FT,
+    ϴ_i::FT,
+    κ_sat_unfrozen::FT,
+    κ_sat_frozen::FT,
+) where {FT}
+    #TBD: can we get rid of this branch? if not: create test for it.
```

Comment:
did we add a test for this?

Reply:
just did to test the 1st branch (we were only testing second branch before)